nemo/collections/nlp/parts/mixins/nlp_adapter_mixins.py

```diff
@@ -88,8 +88,10 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
         self.use_mcore_gpt = hasattr(self, 'mcore_gpt') and self.mcore_gpt
-        if self.use_mcore_gpt:
-            assert HAVE_MEGATRON_CORE, "You set `mcore_gpt` as True but megatron core is not found."
+        self.use_mcore_t5 = hasattr(self, 'mcore_t5') and self.mcore_t5
```

Comment:
Nit: for readability, could you change all instances (including gpt) of `hasattr(self, 'xyz') and self.xyz` to `getattr(self, 'xyz', None)`?

Reply:
Sure. Will fix this.